src/config/field-overrides/program-indicator/OrgUnitField.js

```diff
@@ -87,18 +93,40 @@ export default class OrgUnitField extends Component {
         if (this.hasRequiredParams()) {
             this.fetchDataElements();
         }
+        this.setState({ options: this.getOptions() });
 
         this.updateManualPrevProps();
     }
 
-    componentDidUpdate() {
-        if (
-            this.hasRelevantPropChanged() &&
-            this.shouldFetchDataElementsForProgram()
-        ) {
+    componentDidUpdate(_, prevState) {
+        const isDoneLoading =
+            prevState.loading === true && this.state.loading === false;
+        const hasRelevantPropChanged = this.hasRelevantPropChanged();
+        const requiresDataElementsForProgram = this.requiresDataElementsForProgram();
+        const shouldFetch =
+            hasRelevantPropChanged && requiresDataElementsForProgram;
+        const shouldSetOptions =
+            (isDoneLoading || hasRelevantPropChanged) &&
+            !requiresDataElementsForProgram;
```

Comment:
It's not immediately obvious why `requiresDataElementsForProgram` decides if we should set the options or not. 
And after some more thinking, I'm not sure if we actually need that check? We could just `setOptions` regardless, right? Yeah there's a few more times it would run, but does that matter?

Reply:
You may be correct... But I'm not actually sure.... I think that we probably do need to include `requiresDataElementsForProgram ` in this check.... We need to consider these two cases:
1. The relevant props have changed and no data elements need to be fetched (this could either be because the combination of program+analyticsType doesn't need data elements, or because the relevant data elements had been fetched before and are still available). In this case we're ready to update the options straight away.
2. Or the relevant props changed and for this combination of program+analyticsType we do need to fetch data elements. In this case we need to wait until the data elements are returned from the server, which is an "event" we can identify be the `loading` state transitioning from `true` to `false`

So if I were to write this condition to strictly reflect these two scenarios, I would end up with a slightly different comparison operator, but still including `requiresDataElementsForProgram`:
```js
const shouldSetOptions = (hasRelevantPropChanged && !requiresDataElementsForProgram) || 
     isDoneLoading;
```

But I took a slightly different approach, which I think also makes sense. In the check I try to
1. Did an "event" (aka props or state transition) take place which requires the options to be recomputed?
    ```js
    (isDoneLoading || hasRelevantPropChanged)
    ```
2. And do we have enough information available to actually compute the new `options`?
    ```js
    !requiresDataElementsForProgram
    ```

I actually still think this is a sound approach.

So I'd probably argue that I should keep this check as-is. But change the variable and or method name, because that is what seems to be causing confusion here....

Reply:
> I actually still think this is a sound approach.

> So I'd probably argue that I should keep this check as-is. But change the variable and or method name, because that is what seems to be causing confusion here....

Sounds good to me, it just wasn't immediatly obvious that that `!requiresDataElementsForProgram` is used for ```do we have enough information available to actually compute the new options?```.

Reply:
A compelling argument for keeping the current approach is an edge case we could encounter when moving to the alternative approach I mention above: 
Suppose the user's internet connection is very slow and he/she change the program in the form multiple times in quick succession. If both programs require data elements `loading` will transition from `true` to `false` twice and if we try to set the options after the first transition it'll try producing the options for the second program, but its data elements are not available yet.